image/icns.ksy

```diff
@@ -0,0 +1,889 @@
+              This information can be used to calculate the positions of all elements without having to read the entire data.
+      icon_composer_version_data:
+        doc: The element data for an Icon Composer version number.
+        -webide-representation: '{version:dec}'
+        seq:
+          - id: version
+            type: f4
+            doc: |
+              The version of Icon Composer that created this ICNS file.
+              Some versions of Icon Composer set this to -1 instead of an actual version number.
+      icns_style_packbits:
+        doc: |
+          A run-length encoding compression scheme similar to (but not the same as) PackBits.
+          Used in the RGB and ARGB bitmap icon types.
+        seq:
+          - id: chunks
+            type: chunk
+            repeat: eos
+            doc: The chunks that make up the compressed data.
+        types:
+          chunk:
+            doc: |
+              A single chunk of compressed data.
+              Each chunk stores either a sequence of literal bytes,
+              or a single byte that is repeated a certain number of times.
+            seq:
+              - id: tag
+                type: u1
+                doc: |
+                  The chunk's tag byte,
+                  indicating whether the chunk is a literal or repeat chunk,
+                  as well as how many literal bytes follow or how often the byte should be repeated.
+              - id: literal_data
+                size: tag + 1
+                if: not is_repeat
+                doc: |
+                  If this is a literal chunk,
+                  the literal byte sequence stored in the chunk.
+              - id: repeated_byte
+                type: u1
+                if: is_repeat
+                doc: |
+                  If this is a repeat chunk,
+                  the byte to be repeated in the output.
+            instances:
+              is_repeat:
+                value: tag >= 128
+                doc: |
+                  If true, this is a repeat chunk.
+                  If false, this is a literal chunk.
+              len_literal_data:
+                value: tag + 1
+                if: not is_repeat
+                doc: |
+                  If this is a literal chunk,
+                  the number of literal bytes stored in the chunk.
+              repeat_count:
+                value: tag - 125
+                if: is_repeat
+                doc: |
+                  If this is a repeat chunk,
+                  the number of times the stored byte should be repeated in the output.
+      icon_x1_and_mask_data:
+        doc: The data for a 1-bit monochrome bitmap icon with a 1-bit mask.
+        params:
+          - id: width
+            type: u4
+            doc: The width of the icon in pixels.
+          - id: height
+            type: u4
+            doc: The height of the icon in pixels.
+        seq:
+          - id: icon
+            size: width * height / 8
```

Comment:
`repeat-expr` + `b1`?

Reply:
I thought about it, but decided that it probably wouldn't have any practical use.

* If you're using the spec in a visualizer, having the data parsed into rows/pixels isn't very helpful - if you want to see the bitmap, just look at the hex dump, or maybe the visualizer has a built-in option to render bitmap data.
* If you're using the spec from a programming language, there are probably existing libraries that can read arrays of packed integers much more efficiently than KS can at the moment, or perhaps even image processing libraries that can read raw bitmap data into high-level image objects. In the case of Python that would be `array`, `numpy`, and/or `PIL`/Pillow.